backend/main.py

```
@@ -0,0 +1,312 @@
+async def get_user_profile(liked_songs: str):
+    """Get user profile analysis"""
+    try:
+        # Parse liked songs from query parameter
+        songs_data = json.loads(liked_songs)
+        songs = [Song(**song_data) for song_data in songs_data]
+        
+        user_profile = ml_engine.get_user_profile(songs)
+        return {"user_profile": user_profile}
+    except Exception as e:
+        logger.error(f"User profile error: {e}")
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)